src/types/Currency.sol

```diff
@@ -38,42 +38,36 @@ library CurrencyLibrary {
 
     function transfer(Currency currency, address to, uint256 amount) internal {
         // implementation from
-        // https://github.com/transmissions11/solmate/blob/e8f96f25d48fe702117ce76c79228ca4f20206cb/src/utils/SafeTransferLib.sol
+        // https://github.com/Vectorized/solady/blob/0dde2a008d917aa8076f348eac2855edbe181cc0/src/utils/SafeTransferLib.sol
 
-        bool success;
         if (currency.isNative()) {
+            /// @solidity memory-safe-assembly
             assembly {
-                // Transfer the ETH and store if it succeeded or not.
-                success := call(gas(), to, amount, 0, 0, 0, 0)
+                // Transfer the ETH and revert if it fails.
+                if iszero(call(gas(), to, amount, 0, 0, 0, 0)) {
```

Comment:
this implementation is different from Vectorized implementation, why do you set `argsOffset` and `retOffset` to 0 instead of `codesize()`?

Reply:
They are equivalent as long as the calldata length and returndata length are 0. Using `codesize()` is an optimization for chains without `PUSH0`.